### PR TITLE
xidel: 0.9.6 -> 0.9.8, refactor

### PR DESCRIPTION
**Pull request by jnetod**
###### Motivation for this change

Currently broken:

```
$ nix run github:NixOS/nixpkgs/c28262e2fe67c021c37e2e3532d4ebe77401be81#xidel -- --version
Runtime error 216 at $000000000054BDEB
  $000000000054BDEB
```

I've refactored the expression, now it's based on the source code and not from the binary deb package. Inspiration came from https://github.com/benibela/xidel/blob/Xidel_0.9.8/.travis.yml .

Should help #71689 if #108532 get done.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134267` at 90218b37 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>xidel</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/text/xidel/default.nix:65:3:
```
   |
65 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/tools/text/xidel/default.nix:40:3:
```
   |
40 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/tools/text/xidel/default.nix:73:3:
```
   |
73 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/text/xidel/default.nix [hunk 1]
```diff
@@ -1,48 +1,91 @@
...
+    $out/bin/xidel --version | grep "${version}"
   '';
 
   meta = with lib; {
-    description = "Command line tool to download and extract data from html/xml pages";
-    homepage = "http://videlibri.sourceforge.net/xidel.html";
-    # source contains no license info (AFAICS), but sourceforge says GPLv2
-    license = licenses.gpl2;
-    # more platforms will be supported when we switch to source build
+    description = "Command line tool to download and extract data from HTML/XML pages as well as JSON APIs.";
```
Comment (bjornfor) [inline pkgs/tools/text/xidel/default.nix:86]: Description should not end with dot.

File: pkgs/tools/text/xidel/default.nix [hunk 2]
```diff
@@ -1,48 +1,91 @@
...
+
+  patchPhase = ''
+    patchShebangs build.sh
+    patchShebangs tests/test.sh
+    patchShebangs tests/tests-file-module.sh
+    patchShebangs tests/tests.sh
+    patchShebangs tests/downloadTest.sh
+    patchShebangs tests/downloadTests.sh
+    patchShebangs tests/zorbajsoniq.sh
+    patchShebangs tests/zorbajsoniq/download.sh
```
Comment (SuperSandro2000) [inline pkgs/tools/text/xidel/default.nix:48]:
```suggestion
    patchShebangs \
      build.sh \
      tests/test.sh \
      tests/tests-file-module.sh \
      tests/tests.sh \
      tests/downloadTest.sh \
      tests/downloadTests.sh \
      tests/zorbajsoniq.sh \
      tests/zorbajsoniq/download.sh
```

File: pkgs/tools/text/xidel/default.nix (hunk 1 above)
Comment (jnetod) [reply pkgs/tools/text/xidel/default.nix]: Done.

File: pkgs/tools/text/xidel/default.nix (hunk 2 above)
Comment (jnetod) [reply pkgs/tools/text/xidel/default.nix]: Done.

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Thanks!

File: pkgs/tools/text/xidel/default.nix [hunk 3]
```diff
@@ -1,48 +1,92 @@
...
+  src = fetchFromGitHub {
+    owner = "benibela";
+    repo = pname;
+    rev = "Xidel_${version}";
+    sha256 = "0q75jjyciybvj6y17s2283zis9fcw8w5pfsq8bn7diinnbjnzgl6";
+  };
+
+  nativeBuildInputs = [ fpc ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/text/xidel/default.nix:40]: This should be postPatch.

**Pull request merged**